src/main/java/soot/jimple/toolkits/callgraph/OnFlyCallGraphBuilder.java

```diff
@@ -550,6 +550,25 @@ public void addType(Local receiver, Context srcContext, Type type, Context typeC
           }
         } else {
           VirtualCalls.v().resolve(type, receiver.getType(), site.subSig(), site.container(), targetsQueue, appOnly);
+          if(!targets.hasNext()) {
+            /* In the situation where we find nothing to resolve an invoke to in the first call, this
+             * might be because the type for the invoking object is a abstract class and the method is
+             * declared in a parent class. In this situation, when the abstract class has no classes
+             * that extend it in the scene, resolve would not find any targets for the invoke, even
+             * if the parent contained a possible target. 
+             * 
+             * This may have been by design since without a concrete class, we have no idea if the 
+             * method in the parent class is override. However, the same could be said for any non
+             * private method in the abstract class (and these all resolve fine inside the abstract
+             * class even though there are no sub classes of the abstract class). This makes this
+             * situation a corner case.
+             * 
+             * Where as, it used to not resolve any targets in this situation, I want to at least
+             * resolve the method in the parent class if there is one (as this is technically a
+             * possibility and the only information we have).
+             */
+            VirtualCalls.v().resolveSuperType(type, receiver.getType(), site.subSig(), targetsQueue, appOnly);
```

Comment:
As you mention, the decision to not have any targets for an invitation because we do not know if the invoked method is overridden by a subclass might be a design decision. 

Since this behavior should be configurable to the needs of an analyst, I suggest introducing an option to Soot which by default uses the current behavior and can be used to configure Soot to work as you propose.